lib/xandra/protocol.ex

```diff
@@ -306,6 +306,12 @@ defmodule Xandra.Protocol do
     <<date::32>>
   end
 
+  defp encode_value(:decimal, float) when is_float(float) do
+    encode_value(:decimal, Decimal.new(float))
+  end
+  defp encode_value(:decimal, %Decimal{sign: sign, coef: coef, exp: exp}) do
```

Comment:
Should probably add a guard here so we don't include exponents outside the integer range, even though it's unlikely.